rules/klayout/drc/testing/run_regression.py

```diff
@@ -86,13 +88,13 @@ def call_regression(rule_deck_path, path):
     iname = path.split('.gds')
     if '/' in iname[0]:
         file = iname[0].split('/')
-        os.system(f"klayout -b -r run_{x}_{name_ext}/markers.drc -rd input={path} -rd report={file[-1]}.lyrdb -rd thr={thrCount} {switches} ")
+        check_call(f"klayout -b -r run_{x}_{name_ext}/markers.drc -rd input={path} -rd report={file[-1]}.lyrdb -rd thr={thrCount} {switches} ", shell=True)
     else:
-        os.system(f"klayout -b -r run_{x}_{name_ext}/markers.drc -rd input={path} -rd report={iname[0]}.lyrdb -rd thr={thrCount} {switches} ")
+        check_call(f"klayout -b -r run_{x}_{name_ext}/markers.drc -rd input={path} -rd report={iname[0]}.lyrdb -rd thr={thrCount} {switches} ", shell=True)
 
     marker_gen = []
-    ly = 0
-    remove_if = False
+    ly         = 0
+    remove_if  = False
 
     # Get the small rule deck with gds output
     with open(rule_deck_path, 'r') as f:
```

Comment:
curious why you need to rewrite the drc file?

I think when @mithro suggested having smaller file, he suggested having them maintained in the repo as separate files, not created on the fly by the test script.

Reply:
@proppy Because, the regression needs to change the output to generate polygons to be able to check if the output in the right location or not. Yes, I know that @mithro has suggested to split the files. I already have done that and I could publish them. But the run time will explode as I told @mithro. The expected run time for DRC unit tests would be around 25 hours alone. While now using this current technique is about 30 mins or so.

Reply:
> the regression needs to change the output to generate polygons to be able to check if the output in the right location or not.

Are those check dynamic (i.e: they need to be rewritten everytime you run the test because they depends on the input) or can they be static? if the later maybe you could simply have the rewritten rules as part of the repo test suite rather than regenerating them?

Reply:
@proppy No they can't be static. Because assume you added a new rule or removed a rule how would you know.